docs/DESIGN.md

```diff
@@ -0,0 +1,119 @@
+# Overview
+
+Our goal is to create automation similar to what we have now for dotcom generative SDKs in Go and .NET, except for Enterprise Server and Proxima specifications.
+
+Currently, our go-sdk and dotnet-sdk do not provide support for GitHub Enterprise Server (GHES) or Proxima (dotcom stamp w/ GHEC API surface) in any of their versions. 
+
+We need to extend our SDK compatibility to include enterprise platform API surfaces. By doing so, we can enhance the capabilities we offer to our enterprise customers and strengthen the relationship between our revenue-generating customers and our SDK team.
+
+
+## Acceptance criteria
+
+Our design goals generated SDKs are...
+- Abstract API implementations providing easy developer experience
+- Enable users to easily make use of recommended auth patterns 
+- Include client-side rate-limiting to make workflow execution more reliable
+- Offer complete coverage of the REST API enabling more workflows
+- Be an intuitive and discoverable developer experience
+- Cover all supported versions of both GHES and Proxima
+- Follow the deprecation policies of the GItHub API version at which they are targeted
+- Follow a consistent naming scheme
+- Have a patching plan for changes to the SDK outside of normal API specification updates (e.g. in case of necessary security updates or backports)
+
+
+## Overall our design choice should…
+- Work well with our versioning needs
+- Be reasonable to manage and maintain by the SDK team and our awesome contributors
+- Make the default use case simple to use, understand, and easily differentiated at a glance from the base generative SDK offering
+
+
+## Approach
+
+### Use the generator to propagate language SDKs, storing the core implementation for each language in the generator itself
+
+![generated changes flow](./resources/generated-changes-flow.png)
+
+_These changes represent the product of code generated from [GitHUb's OpenAPI](https://github.com/github/rest-api-description) definitions via [Mirosoft Kiota](https://github.com/microsoft/kiota)._
+
+### Process flow for generation
+1. OpenAPI changes are detected from each API representation
+2. Generator creates the SDKs and ships them to their respective language repos (i.e. dotnet-sdk, go-sdk) and PRs against main
+3. Packages and/or tagged appropriately
```

Comment:
Yeah I struggled with this one because for Go we just Tag and .NET and such there are tags and packages... I prefer your wording and I don't think we have to be pedantic here just for sake of accuracy.